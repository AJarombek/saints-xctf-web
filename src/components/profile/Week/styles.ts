/**
 * JSS styles for the Week component.
 * @author Andrew Jarombek
 * @since 10/19/2020
 */

// That sounds heavenly, would love for you to be there.  But no pressure if you arent feeling up to it.

export default {
  week: {
    display: 'flex'
  }
};
