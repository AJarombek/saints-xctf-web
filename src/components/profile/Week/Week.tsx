/**
 * Component for a week on the calendar.
 * @author Andrew Jarombek
 * @since 10/19/2020
 */

import React from 'react';
import {createUseStyles} from "react-jss";
import styles from "./styles";

interface IProps {}

const useStyles = createUseStyles(styles);

const Week: React.FunctionComponent<IProps> = ({}) => {
    const classes = useStyles();

    return (
        <div className={classes.week}></div>
    );
};

export default Week;
