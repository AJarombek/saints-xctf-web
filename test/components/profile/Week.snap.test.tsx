/**
 * Snapshot test for the Week component.
 * @author Andrew Jarombek
 * @since 3/6/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import Week from '../../../src/components/profile/Week/Week';

describe('Week Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Week
          rangeViewItems={[
            {
              date: '2021-03-02',
              feel: 5,
              miles: 4.5
            },
            {
              date: '2021-03-03',
              feel: 5,
              miles: 5.45
            },
            {
              date: '2021-03-04',
              feel: 5,
              miles: 5.45
            },
            {
              date: '2021-03-05',
              feel: 7,
              miles: 6.3
            },
            {
              date: '2021-03-06',
              feel: 6,
              miles: 6.01
            }
          ]}
          start={moment('2021-03-01')}
          monthStart={moment('2021-03-01')}
          monthEnd={moment('2021-03-31')}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
