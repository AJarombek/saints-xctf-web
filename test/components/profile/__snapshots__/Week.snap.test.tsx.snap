// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week Snapshot Tests renders correctly 1`] = `
<div
  className="week-0-1-1"
  data-cypress="week"
>
  <div
    className="day-0-1-2 currentMonth-0-1-4"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        1
      </p>
    </div>
  </div>
  <div
    className="day-0-1-2 currentMonth-0-1-4 feel-0-1-3 feel-d1-0-1-8"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        2
      </p>
      <div
        className="miles-0-1-6"
      >
        <p>
          4.50
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="day-0-1-2 currentMonth-0-1-4 feel-0-1-3 feel-d2-0-1-9"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        3
      </p>
      <div
        className="miles-0-1-6"
      >
        <p>
          5.45
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="day-0-1-2 currentMonth-0-1-4 feel-0-1-3 feel-d3-0-1-10"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        4
      </p>
      <div
        className="miles-0-1-6"
      >
        <p>
          5.45
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="day-0-1-2 currentMonth-0-1-4 feel-0-1-3 feel-d4-0-1-11"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        5
      </p>
      <div
        className="miles-0-1-6"
      >
        <p>
          6.30
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="day-0-1-2 currentMonth-0-1-4 feel-0-1-3 feel-d5-0-1-12"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        6
      </p>
      <div
        className="miles-0-1-6"
      >
        <p>
          6.01
        </p>
        <p>
          Miles
        </p>
      </div>
    </div>
  </div>
  <div
    className="day-0-1-2 currentMonth-0-1-4"
    data-cypress="day"
  >
    <div>
      <p
        className="dayOfMonth-0-1-5"
      >
        7
      </p>
    </div>
  </div>
  <div
    className="weekTotal-0-1-14"
    data-cypress="weekTotal"
  >
    <div
      className="miles-0-1-15"
    >
      <p>
        27.71
      </p>
      <p>
        Miles
      </p>
    </div>
  </div>
</div>
`;
